Jenkinsfile

```diff
@@ -0,0 +1,99 @@
+// List of targets with supported RF shields to compile
+def targets = [
+  "K64F": ["ATMEL", "MCR20"],
+  //"NUCLEO_F401RE": ["ATMEL", "MCR20"],
+  "NUCLEO_F429ZI": ["ATMEL", "MCR20"],
+  "NCS36510": ["NCS36510"],
+  "UBLOX_EVK_ODIN_W2": ["ATMEL", "MCR20"]
+  ]
+  
+// Map toolchains to compilers
+def toolchains = [
+  ARM: "armcc",
+  GCC_ARM: "arm-none-eabi-gcc"
+  //IAR: "iar_arm"
+  ]
+
+// Supported RF shields
+def radioshields = [
+  "ATMEL",
+  "MCR20",
+  "NCS36510"
+  ]
+
+// Mesh interfaces
+def meshinterfaces = [
+  "6lowpan",
+  "thread"
+  ]
+  
+def stepsForParallel = [:]
+
+// Jenkins pipeline does not support map.each, we need to use oldschool for loop
+for (int i = 0; i < targets.size(); i++) {
+  for(int j = 0; j < toolchains.size(); j++) {
+    for(int k = 0; k < radioshields.size(); k++) {
+      for(int l = 0; l < meshinterfaces.size(); l++) {
+        def target = targets.keySet().asList().get(i)
+        def allowed_shields = targets.get(target)
+        def toolchain = toolchains.keySet().asList().get(j)
+        def compilerLabel = toolchains.get(toolchain)
+        def radioshield = radioshields.get(k)
+        def meshInterface = meshinterfaces.get(l)
+        def stepName = "${target} ${toolchain} ${radioshield} ${meshInterface}"
+        if(allowed_shields.contains(radioshield)) {
+          stepsForParallel[stepName] = buildStep(target, compilerLabel, toolchain, radioshield, meshInterface)
+        }
+      }
+    }
+  }
+}
+
+timestamps {
+  parallel stepsForParallel
+}
+
+def buildStep(target, compilerLabel, toolchain, radioShield, meshInterface) {
+  return {
+    stage ("${target}_${compilerLabel}_${radioShield}_${meshInterface}") {
+      node ("${compilerLabel}") {
+        deleteDir()
+        dir("mbed-os-example-mesh-minimal") {
+          checkout scm
+
+          if ("${radioShield}" == "MCR20") {
+            // Replace default rf shield
+            execute("sed -i 's/\"value\": \"ATMEL\"/\"value\": \"MCR20\"/' mbed_app.json")
```

Comment:
Please note that even though it works in our current environment, using sed with execute step has a possible problem as Windows does not have sed command by default.